engine/src/main/java/org/apache/apex/stram/ContainerRedeployRequest.java

```diff
@@ -0,0 +1,71 @@
+/**
```

Comment:
This class currently is like a "plain old Java object" having no logic. Instead it will be good to move most of the external logic here and add comments. This class can then become the starting point for the new feature and the touchpoints in the rest of the code can be minimized. Examples:

- constructor create the groupId using currentTimeMillis()
- instead of just offering setters the class itself computes the affectedContainers and operatorsToDeploy and operatorsToUndeploy when passed the appropriate arguments.

With proper code comments it becomes easier to follow the class and hence the whole feature.